midend/hsIndexSimplify.h

```diff
@@ -102,6 +102,19 @@ class HSIndexSimplifier : public PassManager {
     }
 };
 
+/// Replaces all ArrayIndex classes with a member classes that have the integer index as string
+/// member.
+/// WARNING: After applying this pass, the IR will not type check any longer.
```

Comment:
This is important. Maybe worth printing a warning when invoking this pass? But this could be considered spam.

Reply:
This does not really make it a midend pass, these are supposed to be mixed and matched, but if you run this you're finished.

Reply:
I can move this back then. It is a utility pass in some form.

Reply:
Did you move it back?